src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/NdkTools.cs

```diff
@@ -0,0 +1,391 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	public abstract class NdkTools
+	{
+		// Target triples used in various places (tools prefix, include directory naming etc)
+		static readonly Dictionary<AndroidTargetArch, string> archTriples = new Dictionary<AndroidTargetArch, string> {
+			{ AndroidTargetArch.Arm, "arm-linux-androideabi" },
+			{ AndroidTargetArch.Arm64, "aarch64-linux-android" },
+			{ AndroidTargetArch.X86, "i686-linux-android" },
+			{ AndroidTargetArch.X86_64, "x86_64-linux-android" },
+		};
+
+		// Architecture names as used by the `platforms/*` directories pre NDK r22
+		static readonly Dictionary<AndroidTargetArch, string> archPlatforms = new Dictionary<AndroidTargetArch, string> {
+			{ AndroidTargetArch.Arm, "arm" },
+			{ AndroidTargetArch.Arm64, "arm64" },
+			{ AndroidTargetArch.X86, "x86" },
+			{ AndroidTargetArch.X86_64, "x86_64" },
+		};
+
+		protected Dictionary<NdkToolKind, string> NdkToolNames = new Dictionary<NdkToolKind, string> {
+			{ NdkToolKind.Assembler, "as" },
+			{ NdkToolKind.Linker, "ld" },
+			{ NdkToolKind.Strip, "strip" },
+		};
+
+		public NdkVersion Version { get; }
+		public string NdkRootDirectory { get; }
+		public bool UsesClang { get; protected set; }
+		public bool NoBinutils { get; protected set; }
+
+		// We can't use MonoAndroidHelper.AndroidSdk.AndroidNdkHostPlatform here since it's
+		// not initialized while running tests and attempts to use cause a NREX
+		// We could call `MonoAndroidHelper.RefreshSupportedVersions` but since this NdkTools
+		// instance already knows the location of NDK, it would be just a waste of time.
+		protected string HostPlatform => GetNdkHostPlatform ();
+		protected bool IsWindows => OS.IsWindows;
+
+		protected TaskLoggingHelper? Log { get; }
+
+		protected NdkTools (string androidNdkPath, NdkVersion version, TaskLoggingHelper? log = null)
```

Comment:
Shouldn't the `Log` property be set to `log`?

Relatedly, do we actually want `log` to be a `TaskLoggingHelper` and not an `Action<TraceLevel, string>`, as is done elsewhere?  That way if `log` is null, we can default it to `Console.WriteLine()`s as a "worst case we need to log *something*" scenario.

Reply:
log is `null` only in some tests and I want to have the full logger available, I've always disliked this bit of the old code as logging of non-error messages was unnecessarily hard.

Reply:
As discussed [on discord](https://discord.com/channels/732297728826277939/732297837953679412/864968631582064671), the first sentence was overlooked:

> Shouldn't the `Log` property be set to `log`?

The `Log` property isn't set by any codepath.

This constructor should be updated to include:

```csharp
Log = log;
```